lib/postgrex/protocol.ex

```diff
@@ -117,14 +125,37 @@ defmodule Postgrex.Protocol do
       types_lock: nil,
       prepare: prepare,
       messages: [],
-      ssl: ssl?,
+      ssl: ssl_opts,
       target_server_type: target_server_type,
       search_path: opts[:search_path]
     }
 
     connect_endpoints(endpoints, sock_opts ++ @sock_opts, connect_timeout, s, status, [])
   end
 
+  defp default_ssl_opts(:verify_full, opts) do
+    case Keyword.fetch(opts, :hostname) do
+      {:ok, hostname} ->
+        [
+          server_name_indication: hostname,
+          customize_hostname_check: [
+            match_fun: :public_key.pkix_verify_hostname_match_fun(:https)
+          ]
```

Comment:
Can we also add `verify: :verify_peer`? I believe it's not enabled by default.

```suggestion
          ],
          verify: :verify_peer
```

Reply:
Sorry, please disregard this, I think it _is_ using verify_peer by default:

If I use just `ssl: :verify_full` I get this error:

```
11:13:24.657 [error] Postgrex.Protocol (#PID<0.203.0>) failed to connect: ** (DBConnection.ConnectionError) ssl connect: Options (or their values) can not be combined: [{verify,verify_peer},
                                                {cacerts,undefined}] - {:options, :incompatible, [verify: :verify_peer, cacerts: :undefined]}
```

Unfortunately I can't just pass `ssl_opts: [cacerts: :public_key.cacerts_get()]` because DO, like I'm sure most servers, are not using public certs? Passing:

```
ssl_opts: [
  cacertfile: "/Users/wojtek/Downloads/ca-certificate.crt"
]
```

that I got from them fixes the error.